src/main/java/com/modus/camel/datasonnet/language/DatasonnetExpression.java

```diff
@@ -57,9 +57,7 @@ public String toString() {
                 expression = script;
                 processor.setDatasonnetScript(script);
             }
-            processor.setInputMimeType(getInputMimeType());
-            processor.setOutputMimeType(getOutputMimeType());
-            Object value = processor.processMapping(exchange);
+            Object value = processor.processMapping(exchange, getInputMimeType(), getOutputMimeType());
```

Comment:
The replaced three lines won't race now, but if the expression was getting called with differing mimetypes across threads and causing a race here, isn't there still a possibility that one call to `configure` followed by `evaluate` can  race another, resulting in one of the evaluates having the wrong mimetypes?

Reply:
hm, that's a good question, I don't believe it's possible to have different mime types across threads, but I definitely need to come up with a good test for this.

Reply:
So the problem wasn't different thread values in DatasonnetExpression.  That guy's values were `null`.  So the NPE I got was from the Processor's code that set the field to a non-null value, then later expected it to be non-null, racing with another thread resetting the value back to null.

If that's not clear, let me know and I'll point out where in the code this happens.